src/Workspaces/Core/Portable/Utilities/AsyncLazy`1.cs

```diff
@@ -520,7 +520,10 @@ private void OnAsynchronousRequestCancelled(object state)
             }
         }
 
-        private sealed class Request
+        /// <remarks>
+        /// This inherits from <see cref="TaskCompletionSource{TResult}"/> to avoid allocating two objects when we can just use one.
+        /// </remarks>
```

Comment:
@CyrusNajmabadi I should comment it better but don't have a particular concern. This is the pattern our codebase had for a few years anyways, and that wasn't a problem. "Clearly" we should add the C++ feature of protected inheritance. I guess I should still comment it further though.

At this point the methods we have are also just mostly helper methods...the two types are already deeply coupled and mistakes could be made easily without such protection.